src/array_partition.jl

```diff
@@ -370,3 +370,29 @@ function LinearAlgebra._swap_rows!(B::ArrayPartition, i::Integer, j::Integer)
     B[i], B[j] = B[j], B[i]
     return B
 end
+
+# linalg mul! overloads for ArrayPartition
+function LinearAlgebra.mul!(C::ArrayPartition, A::ArrayPartition, B::AbstractArray)
```

Comment:
this should probably check for size and either do views or all of `B`?

Reply:
do you mean something like:
```julia
function LinearAlgebra.mul!(C::ArrayPartition{T, S}, A::ArrayPartition{T, S}, B::AbstractArray{T, S}) where T<:Number
```

Reply:
I was out of my mind and it didn't make too much sense.